### DISCUSSION TITLE
Structurizr and Backstage

### DISCUSSION THREAD

**Discussion by simonbrowndotje** [structurizr/java#227] [General]
A few people have asked me about Backstage and Structurizr integration recently. There are various ways to interpret what this means, but I've added a couple of examples at https://github.com/structurizr/examples/tree/main/enterprise

Some related links (many of these relate to generating system landscape diagrams):

- https://engineering.atspotify.com/2022/07/software-visualization-challenge-accepted/
- https://github.com/backstage/backstage/issues/15769
- https://github.com/backstage/backstage/issues/19708
- https://github.com/backstage/backstage/issues/11671
- https://github.com/backstage/backstage/issues/7109
- https://github.com/backstage/backstage/issues/4394

Feel free to add any thoughts on the topic.

**Comment by asos-alexwhite** [comment]
We use backstage and structurizr. Auto layout doesn’t work for us. Our diagrams can be complex  and manually laying them out is necessary to enable them to communicate well. 

We used to export images of diagrams to put into confluence, but we have now created a DSL viewer which looks at the DSL and JSON in source control to render the current view. This makes sure it is as leats up to date. 

We are planning to use iframes in TechDocs to host the same viewer, so that backstage users can get the same up to date diagrams

**Reply by simonbrowndotje** [reply]
Are you syncing entities (e.g. systems) from Backstage into your Structurizr workspace, or vice versa? Or are you simply duplicating them across both?

(this is off-topic, but you can embed live diagrams from a Structurizr on-premises or cloud workspace via an iframe rather than building something yourself; e.g. https://docs.structurizr.com/onpremises/embed and https://docs.structurizr.com/cloud/embed)

**Reply by mawid6** [reply]
@asos-alexwhite - the viewer sounds interesting, is it an off the shelf product or something you created - and if the latter, any chance you would be able to share it or describe what / what approach you used to create it?

**Comment by marksalpeter** [comment]
Hi @simonbrowndotje! I'm really enjoying using structurizr so far. My company uses backstage for their dev hub and the absence of a clean integration with structurizr is noticeable. 

What I think of when I consider a 'backstage integration' is a plugin that works similarly to the backstage techdocs plugin. It would generate a structurizr static site and add host it in an architecture tab inside of a backstage entity page. 

Would anyone else also be interested in a backstage plugin that works this way?


<img width="1617" alt="mockup 1" src="https://github.com/user-attachments/assets/585b656d-731b-4770-9e99-afd9fc3c29b3" />

<img width="1617" alt="mockup 2" src="https://github.com/user-attachments/assets/6eacbd83-a9e8-4c58-893c-e3329cdc89de" />
